common/static/common/js/spec/components/list_spec.js

```diff
@@ -0,0 +1,67 @@
+define(['jquery', 'backbone', 'underscore', 'common/js/components/views/list'],
+    function ($, Backbone, _, ListView) {
+        'use strict';
+        describe('ListView', function () {
+            var Model = Backbone.Model.extend({
+                    defaults: {
+                        name: 'default name'
+                    }
+                }),
+                View = Backbone.View.extend({
+                    tagName: 'div',
+                    className: 'my-view',
+                    template: _.template('<p>Name: "<%- name %>"</p>'),
+                    render: function () {
+                        this.$el.html(this.template(this.model.attributes));
+                        return this;
+                    }
+                }),
+                Collection = Backbone.Collection.extend({
+                    model: Model
+                }),
+                expectListNames = function (names) {
+                    expect(listView.$('.my-view').length).toBe(names.length);
+                    _.each(names, function (name, index) {
+                        expect($(listView.$('.my-view')[index]).text()).toContain(name);
+                    });
+                },
+                listView;
+
+            beforeEach(function () {
+                setFixtures('<div class="list"></div>');
+                listView = new ListView({
+                    el: $('.list'),
+                    collection: new Collection(
+                        [{name: 'first model'}, {name: 'second model'}, {name: 'third model'}]
+                    ),
+                    itemViewClass: View
+                });
+                listView.render();
+            });
+
+            it('renders itself', function () {
+                expect(listView.$('.my-view').length).toBe(3);
+                expect($(listView.$('.my-view')[0]).text()).toContain('first model');
+                expect($(listView.$('.my-view')[1]).text()).toContain('second model');
+                expect($(listView.$('.my-view')[2]).text()).toContain('third model');
+            });
+
+            it('re-renders itself when the collection changes', function () {
+                var model = new Model({name: 'fourth model'});
+                expectListNames(['first model', 'second model', 'third model']);
+                listView.collection.add([model]);
+                expectListNames(['first model', 'second model', 'third model', 'fourth model']);
+                listView.collection.remove(model);
```

Comment:
I recommend having tests for adding and removing from the front and from the middle of the list. I know that the current algorithm is naive and re-renders everything, but it will ensure that any future algorithm also behaves correctly. I would separate out add and remove into their own tests at the same time.